app/src/main/java/com/nervousfish/nervousfish/modules/cryptography/EncryptedSaver.java

```diff
@@ -0,0 +1,266 @@
+     * @param pass  THe string to encrypt
+     * @return  The encrypted string.
+     */
+    public static String hashWithoutSalt(final String pass) {
+        try {
+            final MessageDigest digest = MessageDigest.getInstance("SHA-256");
+            digest.reset();
+            final byte[] hash = digest.digest(pass.getBytes(StandardCharsets.UTF_8));
+            LOGGER.info("Hashed the pass with SHA-256, no salt");
+            return new String(hash, UTF_8);
+        } catch (NoSuchAlgorithmException e) {
+            LOGGER.error("SHA-256 is not a valid encryptionalgorithm", e);
+        } catch (UnsupportedEncodingException e) {
+            LOGGER.error("UTF-8 is not a valid encoding method", e);
+        }
+        return null;
+    }
+
+    /**
+     * Hashes a pass to an encrypted string with a salt bytestring.
+     * @param salt The salt bytestring to encrypt the pass with.
+     * @param pass  The string to encrypt
+     * @return The encrypted string.
+     */
+    /*
+    public static String hashUsingSalt(byte[] salt, String pass){
+        try {
+            final MessageDigest digest = MessageDigest.getInstance("SHA-256");
+            digest.reset();
+            digest.update(salt);
+            final byte[] hash = digest.digest(pass.getBytes(StandardCharsets.UTF_8));
+            LOGGER.info("Hashed the pass using the salt");
+            return new String(hash, "UTF-8");
+        } catch (NoSuchAlgorithmException e){
+            LOGGER.error("SHA-256 is not a valid encryptionalgorithm", e);
+        } catch (UnsupportedEncodingException e) {
+            LOGGER.error("UTF-8 is not a valid encoding method", e);
+        }
+        return null;
+    }
+    */
+
+    /**
+     * Make a key for encryption based on a password.
+     * @param password The password to make the key with
+     * @return The SecretKey based on the password.
+     */
+    public static SecretKey makeKey(final String password) throws InvalidKeySpecException {
+        LOGGER.info("Making key for encryption");
+        try {
+            final PBEKeySpec keySpec = new PBEKeySpec(password.toCharArray());
+            final SecretKeyFactory keyFactory = SecretKeyFactory.getInstance(PBE_WITH_MD5_AND_DES);
+            final SecretKey key = keyFactory.generateSecret(keySpec);
+            LOGGER.info("Key successfully made");
+            return key;
+        } catch (NoSuchAlgorithmException e) {
+            LOGGER.error("There isn't an algorithm like PBEWithMD5AndDES", e);
+        }
+        return null;
+    }
+
+    /**
+     * Encrypts or decrypts a string with a password to
+     * @param toEncrypt The string to be encrypted/decrypted
+     * @param key  The secret key made from the password.
+     * @param encrypt   whether we're encrypting or decrypting.
+     * @return  The encrypted/decrypted bytearray.
+     */
+    public static String encryptOrDecryptWithPassword(final String toEncrypt, final SecretKey key, final boolean encrypt)
```

Comment:
Would be even better if you removed the method makeKey, because the client now has to call 2 methods. First makeKey, the encryptOrDecryptWithPassword. As a client I would like to use your library by calling encrypt(byte[]) and decrypt(byte[]), because then I only have to use a single function, the code needed to generate a SecretKey becomes simpler (see https://stackoverflow.com/questions/14204437/convert-byte-array-to-secret-key) and because the password turns out to be a String accidentally, but the rhythm and visual verification are both other data types - and all data types can be turned into byte array easily (eg "asdf'.getBytes())